src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -1555,6 +1555,159 @@ class DropMaterializedViewStatementSegment(BaseSegment):
+        "DATABASE",
+        Ref("DatabaseReferenceSegment"),
+        Sequence("WITH", optional=True),
+        AnyNumberOf(
+            Sequence(
+                "OWNER",
+                Ref("EqualsSegment", optional=True),
+                Ref("ObjectReferenceSegment"),
+            ),
+            Sequence(
+                "TEMPLATE",
+                Ref("EqualsSegment", optional=True),
+                Ref("ObjectReferenceSegment"),
+            ),
+            Sequence(
+                "ENCODING",
+                Ref("EqualsSegment", optional=True),
+                OneOf(Ref("QuotedLiteralSegment"), "DEFAULT"),
+            ),
+            OneOf(
+                Sequence(
+                    "LOCALE",
+                    Ref("EqualsSegment", optional=True),
+                    Ref("QuotedLiteralSegment"),
+                ),
+                Sequence(
+                    Sequence(
+                        "LC_COLLATE",
+                        Ref("EqualsSegment", optional=True),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                    Sequence(
+                        "LC_CTYPE",
+                        Ref("EqualsSegment", optional=True),
+                        Ref("QuotedLiteralSegment"),
+                    ),
+                ),
+            ),
+            Sequence(
+                "TABLESPACE",
+                Ref("EqualsSegment", optional=True),
+                Ref("ParameterNameSegment"),
+            ),
+            Sequence(
+                "ALLOW_CONNECTIONS",
+                Ref("EqualsSegment", optional=True),
+                Ref("BooleanLiteralGrammar"),
+            ),
+            Sequence(
+                "CONNECTION",
+                "LIMIT",
+                Ref("EqualsSegment", optional=True),
+                Ref("NumericLiteralSegment"),
+            ),
+            Sequence(
+                "IS_TEMPLATE",
+                Ref("EqualsSegment", optional=True),
+                Ref("BooleanLiteralGrammar"),
+            ),
+        ),
+    )
+
+
+@postgres_dialect.segment()
+class AlterDatabaseStatementSegment(BaseSegment):
+    """A `ALTER DATABASE` statement.
+
+    As specified in https://www.postgresql.org/docs/14/sql-alterdatabase.html
+    """
+
+    type = "alter_database_statement"
+
+    match_grammar = StartsWith(Sequence("ALTER", "DATABASE"))
+
+    parse_grammar = Sequence(
+        "ALTER",
+        "DATABASE",
+        Ref("DatabaseReferenceSegment"),
+        OneOf(
+            Sequence(
+                Sequence("WITH", optional=True),
+                AnyNumberOf(
+                    Sequence("ALLOW_CONNECTIONS", Ref("BooleanLiteralGrammar")),
+                    Sequence(
+                        "CONNECTION",
+                        "LIMIT",
+                        Ref("NumericLiteralSegment"),
+                    ),
+                    Sequence("IS_TEMPLATE", Ref("BooleanLiteralGrammar")),
+                ),
```

Comment:
```suggestion
                AnyNumberOf(
                    Sequence("ALLOW_CONNECTIONS", Ref("BooleanLiteralGrammar")),
                    Sequence(
                        "CONNECTION",
                        "LIMIT",
                        Ref("NumericLiteralSegment"),
                    ),
                    Sequence("IS_TEMPLATE", Ref("BooleanLiteralGrammar")),
                    min_times=1,
                ),
```
slight nuance here, we can have any number of these but we must have at least one of them so we use the min_times argument to AnyNumberOf